### DISCUSSION TITLE
Generate new sequences

### DISCUSSION THREAD

**Discussion by aqred1** [facebookresearch/esm#70] [Q&A] [answered]
Hey, thanks for this great work.

How can we explore the sequences that the pre-trained model would generate? I assume by feeding sequences with all positions masked? But that would require defining the length of the sequence a priori, and also, how well would it work since the models were trained by masking 15% of the AA??

**Comment by tomsercu** [comment, marked as answer]
Masking everything won't give your LM any reasonable input to go off, I wouldn't expect really anything reasonable to happen that way.
You could look at Gibbs sampling from a language model, like in this paper https://www.biorxiv.org/content/10.1101/2021.01.26.428322v1

**Reply by aqred1** [reply]
Thanks, that helps
